wireguard-go fails to compile on darwin

I can't compile `wireguard-go` on macOS:

```
> nix-build -A wireguard-go '<nixpkgs>'
these derivations will be built:
  /nix/store/0482j0x2mjwaqy8wbl122k8617rlx18j-wireguard-go-0.0.20190517.drv
building '/nix/store/0482j0x2mjwaqy8wbl122k8617rlx18j-wireguard-go-0.0.20190517.drv'...
unpacking sources
unpacking source archive /nix/store/vshdawjvk9hh8ab409ab39lxlqixjq3x-source
source root is source
patching sources
configuring
unpacking source archive /nix/store/hj7arpxk70k8qqsq568g62k3wacnyfpd-go-winio
unpacking source archive /nix/store/a511h57qlfkb5k9a99xz2fiwwl98gas6-errors
unpacking source archive /nix/store/4icd0gnfcvq9drcs47d5pg0bmm1qj70c-crypto-a29dc8f
unpacking source archive /nix/store/7iaxclf4q747qvdaihg5q2mf2xmbhpq2-net-7f726ca
unpacking source archive /nix/store/0apysc7pr5vmjhs1asqah41mi1iqxrps-sys-a43fa87
unpacking source archive /nix/store/1swr6745l26rx0k5qyvizjnpqfmcddi3-text
building
go/src/git.zx2c4.com/wireguard-go/main.go:18:2: cannot find package "golang.zx2c4.com/wireguard/device" in any of:
        /nix/store/46q050s2wny5kmp1yssk1bsszwykg523-go-1.12.9/share/go/src/golang.zx2c4.com/wireguard/device (from $GOROOT)
        /private/var/folders/bg/mvwkzdvx2gj0ymqcd_8_4vp00000gn/T/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/device (from $GOPATH)
go/src/git.zx2c4.com/wireguard-go/main.go:19:2: cannot find package "golang.zx2c4.com/wireguard/ipc" in any of:
        /nix/store/46q050s2wny5kmp1yssk1bsszwykg523-go-1.12.9/share/go/src/golang.zx2c4.com/wireguard/ipc (from $GOROOT)
        /private/var/folders/bg/mvwkzdvx2gj0ymqcd_8_4vp00000gn/T/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/ipc (from $GOPATH)
go/src/git.zx2c4.com/wireguard-go/main.go:20:2: cannot find package "golang.zx2c4.com/wireguard/tun" in any of:
        /nix/store/46q050s2wny5kmp1yssk1bsszwykg523-go-1.12.9/share/go/src/golang.zx2c4.com/wireguard/tun (from $GOROOT)
        /private/var/folders/bg/mvwkzdvx2gj0ymqcd_8_4vp00000gn/T/nix-build-wireguard-go-0.0.20190517.drv-0/go/src/golang.zx2c4.com/wireguard/tun (from $GOPATH)
builder for '/nix/store/0482j0x2mjwaqy8wbl122k8617rlx18j-wireguard-go-0.0.20190517.drv' failed with exit code 1
error: build of '/nix/store/0482j0x2mjwaqy8wbl122k8617rlx18j-wireguard-go-0.0.20190517.drv' failed
```


**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3`
 - channels(eridius): `"nixpkgs-20.03pre194957.bef773ed53f"`
 - nixpkgs: `/Users/eridius/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: wireguard-go
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/tools/networking/wireguard-go/default.nix
+++ b/pkgs/tools/networking/wireguard-go/default.nix
@@ -2,13 +2,13 @@
 
 buildGoPackage rec {
   pname = "wireguard-go";
-  version = "0.0.20190517";
+  version = "0.0.20190908";
 
-  goPackagePath = "git.zx2c4.com/wireguard-go";
+  goPackagePath = "golang.zx2c4.com/wireguard";
 
   src = fetchzip {
     url = "https://git.zx2c4.com/wireguard-go/snapshot/wireguard-go-${version}.tar.xz";
-    sha256 = "0ir3dp13vkkqr76q0jvw610qw40053ngk51psqhqxfaw3jicdqgr";
+    sha256 = "1jsch0157nk86krdknij7dsvg6i7ar0ydhy07r40drhxqyp3q0hx";
   };
 
   goDeps = ./deps.nix;
@@ -20,6 +20,5 @@ buildGoPackage rec {
     homepage = https://git.zx2c4.com/wireguard-go/about/;
     license = licenses.gpl2;
     maintainers = with maintainers; [ elseym kirelagin yegortimoshenko zx2c4 ];
-    platforms = platforms.darwin;
   };
 }

--- a/pkgs/tools/networking/wireguard-go/deps.nix
+++ b/pkgs/tools/networking/wireguard-go/deps.nix
@@ -1,30 +1,12 @@
 # file generated from go.mod using vgo2nix (https://github.com/adisbladis/vgo2nix)
 [
   {
-    goPackagePath = "github.com/Microsoft/go-winio";
-    fetch = {
-      type = "git";
-      url = "https://github.com/Microsoft/go-winio";
-      rev = "v0.4.12";
-      sha256 = "10v2f1xaw2cc97mjqnxzgs9ydpqv71f0ynp1spcywqw97la56zqw";
-    };
-  }
-  {
-    goPackagePath = "github.com/pkg/errors";
-    fetch = {
-      type = "git";
-      url = "https://github.com/pkg/errors";
-      rev = "v0.8.1";
-      sha256 = "0g5qcb4d4fd96midz0zdk8b9kz8xkzwfa8kr1cliqbg8sxsy5vd1";
-    };
-  }
-  {
     goPackagePath = "golang.org/x/crypto";
     fetch = {
       type = "git";
       url = "https://go.googlesource.com/crypto";
-      rev = "a29dc8fdc734";
-      sha256 = "07f0jj4haqs0ywc1akk1qjwn5msl2j0pry1rxjkkbfcq4r6ihc1p";
+      rev = "9756ffdc2472";
+      sha256 = "0q7hxaaq6lp0v8qqzifvysl47z5rfdlrxkh3d29vsl3wyby3dxl8";
     };
   }
   {
@@ -32,8 +14,8 @@
     fetch = {
       type = "git";
       url = "https://go.googlesource.com/net";
-      rev = "7f726cade0ab";
-      sha256 = "0zv5var5agbhd2mcn7lciwip0j49gxka73f0csvh3hvb1di068gn";
+      rev = "ba9fcec4b297";
+      sha256 = "1hbqvy6r0s5h0dpdqw8fynl3cq0acin3iyqki9xvl5r8h33yb9bx";
     };
   }
   {
@@ -41,8 +23,8 @@
     fetch = {
       type = "git";
       url = "https://go.googlesource.com/sys";
-      rev = "a43fa875dd82";
-      sha256 = "0xvwwaqcj02a9n8izs4zrlg8vvz7wffli955i93nc34m1f0w8vyg";
+      rev = "19e00faab6ad";
+      sha256 = "0znjx9xbyjdvk8cafdc1jl22gg29asxl5yl7mi3yz6zx3h39rm8y";
     };
   }
   {
@@ -50,8 +32,17 @@
     fetch = {
       type = "git";
       url = "https://go.googlesource.com/text";
-      rev = "v0.3.0";
-      sha256 = "0r6x6zjzhr8ksqlpiwm5gdd7s209kwk5p4lw54xjvz10cs3qlq19";
+      rev = "v0.3.2";
+      sha256 = "0flv9idw0jm5nm8lx25xqanbkqgfiym6619w575p7nrdh0riqwqh";
+    };
+  }
+  {
+    goPackagePath = "golang.org/x/tools";
+    fetch = {
+      type = "git";
+      url = "https://go.googlesource.com/tools";
+      rev = "90fa682c2a6e";
+      sha256 = "03ic2xsy51jw9749wl7gszdbz99iijbd2bckgygl6cm9w5m364ak";
     };
   }
 ]

